src/linesearch/line_model.jl

```diff
@@ -112,5 +122,20 @@ i.e.,
 """
 function hess(f::LineModel, t::AbstractFloat)
   NLPModels.increment!(f, :neval_hess)
-  return dot(f.d, hprod(f.nlp, f.x + t * f.d, f.d))
+  @. f.xt = f.x + t * f.d
+  return dot(f.d, hprod(f.nlp, f.xt, f.d))
```

Comment:
Shouldn’t this implement the NLPModels API properly, i.e., with `hess_coord`, etc? It should be easy; the Hessian is just a scalar here.

Reply:
It can't implement the API properly because of the arguments being scalars, so even if we implement the missing functions, they can't be tested or used as NLPs. Alternatively, we can change to use a proper NLPModel, similar to what I tried to do for the Merit function in #145. We would essentially remove this `LineModel` and use `MeritModels` instead, like in #146. If we decide to go this way I can try to make a simplified version of #145 and #146.